src/app/pages/test-report/cat-b/test-report.cat-b.page.ts

```diff
@@ -1,54 +1,278 @@
+import { EtaInvalidModal } from '@pages/test-report/components/eta-invalid-modal/eta-invalid-modal';
+import { EndTestModal } from '@pages/test-report/components/end-test-modal/end-test-modal';
+import { OverlayCallback } from '../test-report.model';
+import { ModalEvent } from '../test-report.constants';
+import { isRemoveFaultMode, isSeriousMode, isDangerousMode } from '../test-report.selector';
+import { getTestReportState } from '../test-report.reducer';
+import {
+  TestReportViewDidEnter,
+  CalculateTestResult,
+  TerminateTestFromTestReport,
+} from '../test-report.actions';
+import { CAT_B } from '../../page-names.constants';
 
-interface CatBTestReportPageState {}
-
-type TestReportPageState = CommonTestReportPageState & CatBTestReportPageState;
+interface TestReportPageState {
+  candidateUntitledName$: Observable<string>;
+  isRemoveFaultMode$: Observable<boolean>;
+  isSeriousMode$: Observable<boolean>;
+  isDangerousMode$: Observable<boolean>;
+  manoeuvres$: Observable<boolean>;
+  testData$: Observable<CatBUniqueTypes.TestData>;
+  testRequirements$: Observable<CatBUniqueTypes.TestRequirements>;
+}
 
 @Component({
-  selector: 'app-test-report-cat-b',
-  templateUrl: './test-report.cat-b.page.html',
-  styleUrls: ['./test-report.cat-b.page.scss'],
+  selector: '.test-report-cat-b-page',
+  templateUrl: 'test-report.cat-b.page.html',
+  styleUrls: ['test-report.cat-b.page.scss'],
 })
 export class TestReportCatBPage extends TestReportBasePageComponent implements OnInit {
 
   pageState: TestReportPageState;
+  subscription: Subscription;
+  competencies = Competencies;
+  legalRequirements = LegalRequirements;
+  eta = ExaminerActions;
+  displayOverlay: boolean;
+
+  isRemoveFaultMode: boolean = false;
+  isSeriousMode: boolean = false;
+  isDangerousMode: boolean = false;
+  manoeuvresCompleted: boolean = false;
+  isTestReportValid: boolean = false;
+  isEtaValid: boolean = true;
+
+  modal: HTMLIonModalElement;
+  missingLegalRequirements: legalRequirementsLabels[] = [];
 
   constructor(
-    private navController: NavController,
-    public routeByCat: RouteByCategoryProvider,
-    store$: Store<StoreModel>,
     platform: Platform,
     authenticationProvider: AuthenticationProvider,
     router: Router,
+    store$: Store<StoreModel>,
+    private modalController: ModalController,
+    public testReportValidatorProvider: TestReportValidatorProvider,
+    public screenOrientation: ScreenOrientation,
+    public insomnia: Insomnia,
+    public statusBar: StatusBar,
   ) {
-    super(store$, platform, authenticationProvider, router);
+    super(platform, authenticationProvider, router, store$);
+    this.displayOverlay = false;
+  }
+
+  getCallback(): OverlayCallback {
+    return {
+      callbackMethod: () => {
+        this.toggleReportOverlay();
+      },
+    };
   }
 
-  ngOnInit() {
+  ngOnInit(): void {
     super.onInitialisation();
+    // this.pageState = {
+    //   ...this.commonPageState,
+    // };
+
```

Comment:
could this be removed?

Reply:
will be required shortly when i move common functionality to basepage so leaving there for now